models/surgical_tool_segmentation/docs/readme.md

```diff
@@ -0,0 +1,95 @@
+# Description
+A pre-trained model for the surgical tool segmentation task.
+
+# Model Overview
+This model is trained using a flexible unet structure with an efficient-b0 [1] as the backbone and a UNet architecture [2] as the decoder. Datasets use private samples from [Activ Surgical](https://www.activsurgical.com/).
+The [pytorch model](https://drive.google.com/file/d/14r6WmzaZrgaWLGu0O9vSAzdeIGVFQ3cs/view?usp=sharing) and [torchscript model](https://drive.google.com/file/d/1i-e5xXHtmvmqitwUP8Q3JqvnmN3mlrEm/view?usp=sharing) are shared in google drive. Details can be found in large_files.yml file.
+
+## Data
+Datasets used in this work were provided by [Activ Surgical](https://www.activsurgical.com/).
+
+Since datasets are private, existing public datasets like [EndoVis 2017](https://endovissub2017-roboticinstrumentsegmentation.grand-challenge.org/Data/) can be used to train a similar model.
+
+## Training configuration
+The training was performed with an at least 12GB-memory GPU.
+
+Actual Model Input: 736 x 480 x 3
+
+## Input and output formats
+Input: 3 channel video frames
+
+Output: 2 channels: Label 1: tools; Label 0: everything else
+
+## Scores
+This model achieves the following IoU score on the test dataset (our own split from the first batch data):
+
+Mean IoU = 0.87
+
+## commands example
+Execute training:
+
+```
+python -m monai.bundle run training --meta_file configs/metadata.json --config_file configs/train.json --logging_file configs/logging.conf
+```
+
+Override the `train` config to execute evaluation with the trained model:
+
+```
+python -m monai.bundle run evaluating --meta_file configs/metadata.json --config_file "['configs/train.json','configs/evaluate.json']" --logging_file configs/logging.conf
+```
+
+Execute inference:
+
+```
+python -m monai.bundle run evaluating --meta_file configs/metadata.json --config_file configs/inference.json --logging_file configs/logging.conf
+```
+
+Export checkpoint to TorchScript file:
+
+```
+python -m monai.bundle ckpt_export network_def --filepath models/model.ts --ckpt_file models/model.pt --meta_file configs/metadata.json --config_file configs/inference.json
+```
+
+Export checkpoint to onnx file, which has been tested on pytorch 1.12.0:
+
+```
+python scripts/export_to_onnx.py --model models/model.pt --outpath models/model.onnx
+```
+
+Export TorchScript file to a torchscript module targeting a TensorRT engine with float16 precision.
+
+```
+torchtrtc -p f16 models/model.ts models/model_trt.ts "[(1,3,736,480);(4,3,736,480);(8,3,736,480)]"
```

Comment:
Following the advice of https://github.com/pytorch/TensorRT#nvidia-ngc-container, and using the container `nvidia/pytorch:22.09-py3`, still encountered error 
> torchtrtc -p f16 models/model.ts models/model_trt.ts "[(1,3,736,480);(4,3,736,480);(8,3,736,480)]"
> terminate called after throwing an instance of 'c10::Error' 
>  what():  Trying to create tensor with negative dimension -1: [-1, 3, 736, 480]

Are there any additional steps within the container needed?

Reply:
Hi @jin-nvidia ,
     This error may caused by hardware and software problems. You can add "-v" option first to check the detail.
Thanks,
Bin